### DISCUSSION TITLE
[General] <Title> Set up a archive node based on the light node

### DISCUSSION THREAD

**Discussion by corerouter** [zkSync-Community-Hub/zksync-developers#701] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Now I have setup a light node according to https://github.com/matter-labs/zksync-era/blob/main/docs/guides/external-node/00_quick_start.md and then download the latest zksync pgdump file. 

How can I set up a archive node with the latest pgdump on top of the light node?

### Repo Link (Optional)

https://github.com/matter-labs/zksync-era

**Comment by alhenaai** [comment]
To set up an archive node based on a light node in ZKsync, you'll need to follow these steps after setting up your light node and downloading the latest ZKsync pgdump file:

1. **Stop the ZKsync Node**: Ensure that the ZKsync node is not running before you proceed with the setup of the archive node.

2. **Remove SK Cache**: Delete the SK cache related to your node. This corresponds to the `EN_STATE_CACHE_PATH`.

3. **Remove Your Current Database**: Clear out the existing database to prepare for the restoration with the new pgdump.

4. **Restore with the New Dump**: Use the pgdump file you downloaded to restore your database. This step is crucial as it populates your database with the complete historical data required for an archive node.

5. **Start the ZKsync Node**: Once the restoration is complete, you can start your ZKsync node. It will now function as an archive node, providing access to the entire history of the blockchain.

6. **Monitor Node Behavior**: After starting the node, monitor its behavior to ensure it is functioning correctly as an archive node. You can find more details on monitoring in the observability section of the ZKsync documentation.

For detailed instructions and configurations, you can refer to the [Running a Node](https://docs.zksync.io/zksync-node/running-node) documentation on the ZKsync official documentation site.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by popzxc** [comment, marked as answer]
Hey @corerouter. Instructions on running archive node are provided in the [ansible-en-role](https://github.com/matter-labs/ansible-en-role) repo ([there is a link](https://github.com/matter-labs/zksync-era/blob/main/docs/guides/external-node/00_quick_start.md#advanced-setup) on the "Quick Start" page in docs).
